src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -78,10 +83,11 @@ abstract class AbstractCheckTask(
     protected val sessionKey: SessionKey,
     private val parentEventID: EventID,
     private val messageStream: Observable<StreamContainer>,
-    private val eventBatchRouter: MessageRouter<EventBatch>
+    private val eventBatchRouter: MessageRouter<EventBatch>,
+    private val resultsStorage: ResultsStorage
```

Comment:
I think it will be better if we move it into the rule configuration. We can hide ID knowledge from rule. We can pass the lambda `(EventStatus) -> Unit` instead of `ResultsStorage`